iotdevice/transport/http/http.go

```diff
@@ -124,21 +127,166 @@ func (tr *Transport) UpdateTwinProperties(ctx context.Context, payload []byte) (
 	return 0, ErrNotImplemented
 }
 
-//CreateOrUpdateModuleIdentity Creates or updates the module identity for a device in the IoT Hub.
-//Notice the method is PUT and overrides previous data.
-func (tr *Transport) CreateOrUpdateModuleIdentity(ctx context.Context, identity *common.ModuleIdentity) error {
```

Comment:
What's the reason of this method removal? It's going to affect this package users.

Reply:
Multiple reasons:

- The name is confusing; it doesn't update any module identity, but just the  `edgeAgent` one;
- there's no test that checks that this method actually works;
- the API is calling is the same as the iotservice one; for consistency, we should have the same methods. https://github.com/amenzhinsky/iothub/blob/df8442dfcacaa601813dd7be41ec92017ab71c57/iotservice/client.go#L872-L953
- the `ModuleIdentity` struct is not very convenient to use as it has a [nested definition](https://github.com/amenzhinsky/iothub/blob/df8442dfcacaa601813dd7be41ec92017ab71c57/common/message.go#L73-L83) (as opposed to using the `Authentication` type), like it's done in `iotservice`. https://github.com/amenzhinsky/iothub/blob/df8442dfcacaa601813dd7be41ec92017ab71c57/iotservice/registry.go#L57
- related to the above, but not that important, having `ModuleIdentity` in the `message.go` file feels somewhat odd.